test/puppetlabs/puppetdb/integration/smoke.clj

```diff
@@ -27,3 +27,30 @@
           (is (= catalog-uuid report-uuid)))))))
 
 
+(deftest ^:integration rich-data-agent-run
+  (with-open [pg (int/setup-postgres)
+              pdb (int/run-puppetdb pg {})
+              ps (int/run-puppet-server [pdb] {:agent {:rich-data true}})]
+
+    (testing "Agent run succeeds"
+      (let [{:keys [out]} (int/run-puppet-as "my-agent" ps pdb "notify { 'foo':\n  message => {\n  'regex'     => /[a-z]+/,\n  'object'    => Object[{name => 'FindMe', 'attributes' => {'x' => String}}],\n  'version'   => SemVer('3.4.1'),\n  'semver'    => SemVerRange('>=3.4.1'),\n  'timestamp' => Timestamp('1994-03-25T19:30:00'),\n  'timespan'  => Timespan('3-10:00:00'),\n  }\n}" {:rich-data true})]
+        (is (re-find #">=3.4.1" out))))
```

Comment:
I'm thinking we might want to add more checks here for the other rich data objects too, but I can help add these if needed.